internal/runner/rspec.go

```diff
@@ -112,3 +113,56 @@ func (r Rspec) commandNameAndArgs(testCases []string) (string, []string, error)
 	words = slices.Replace(words, idx, idx+1, testCases...)
 	return words[0], words[1:], nil
 }
+
+// RspecExample represents a single test example in an Rspec report.
+type RspecExample struct {
+	Id              string  `json:"id"`
+	Description     string  `json:"description"`
+	FullDescription string  `json:"full_description"`
+	Status          string  `json:"status"`
+	FilePath        string  `json:"file_path"`
+	LineNumber      int     `json:"line_number"`
+	RunTime         float64 `json:"run_time"`
+}
+
+// RspecReport is the structure for Rspec JSON report.
+type RspecReport struct {
+	Version  string         `json:"version"`
+	Seed     int            `json:"seed"`
+	Examples []RspecExample `json:"examples"`
+}
+
+// GetExamples returns an array of test examples within the given files.
+func (r Rspec) GetExamples(files []string) ([]plan.TestCase, error) {
+	cmdName, cmdArgs, err := r.commandNameAndArgs(files)
+	if err != nil {
+		return nil, err
+	}
+
+	cmdArgs = append(cmdArgs, "--dry-run", "--format", "json")
```

Comment:
This would use the user defined test command for the dry run right?
I don't know how `--dry-run` works with other rspec options (say if a user provided more option flags). Would the default rspec command `bundle exec rspec {files} --dry-run` be more robust?

Reply:
Yes, it will use the exact same command but with `--dry-run`. The dry run is basically similar to running Rspec with all options, but without actually executing the test. I would argue this is more robust because the dry run result will reflect the actual Rspec run. If, for instance, we use the default `bundle exec rspec {files} --dry-run`, but the customers use `--tag` to exclude some tests, thenm the excluded tests will be in the dry run result.

Reply:
oh yeah! I forget about things like `tags`! Good call, we should keep using the user provided command